packages/react/src/NavList/NavList.tsx

```diff
@@ -259,7 +259,8 @@ const Group: React.FC<NavListGroupProps> = ({title, children, sx: sxProp = defau
     <>
       {/* Hide divider if the group is the first item in the list */}
       <ActionList.Divider sx={{'&:first-child': {display: 'none'}}} />
-      <ActionList.Group {...props} title={title} sx={sxProp}>
+      <ActionList.Group {...props} sx={sxProp}>
+        <ActionList.GroupHeading as="h3">{title}</ActionList.GroupHeading>
```

Comment:
Updated here to make the tests pass but I thought this should be a separate PR (at least the as prop) so pushed one https://github.com/primer/react/pull/4593. What ever gets merge first, we need to resolve the conflict but should be very straightforward 🙌🏻